content/en/docs/upgrade/Air-Gapped-Upgrade-with-KubeKey.md

```diff
@@ -0,0 +1,347 @@
+
+{{</ notice >}}
+
+
+### Step 1: Download KubeKey
+
+Similar to installing KubeSphere on Linux in an online environment, you also need to [download KubeKey](https://github.com/kubesphere/kubekey/releases) first. Download the `tar.gz` file, and transfer it to your local machine which serves as the taskbox for installation. After you uncompress the file, execute the following command to make `kk` executable:
+
+```bash
+chmod +x kk
+```
+
+### Step 2: Prepare Installation Images
+
+As you install KubeSphere and Kubernetes on Linux, you need to prepare an image package containing all the necessary images and download the Kubernetes binary file in advance.
+
+1. Download the image list file `images-list.txt` from a machine that has access to the Internet through the following command:
+
+   ```bash
+   curl -L -O https://github.com/kubesphere/ks-installer/releases/download/v3.0.0/images-list.txt
+   ```
+
+   {{< notice note >}}
+
+   This file lists images under `##+modulename` based on different modules. You can add your own images to this file following the same rule. To view the complete file, see [Appendix](../air-gapped-installation/#image-list-of-kubesphere-v300).
+
+   {{</ notice >}} 
+
+2. Download `offline-installation-tool.sh`.
+
+   ```bash
+   curl -L -O https://github.com/kubesphere/ks-installer/releases/download/v3.0.0/offline-installation-tool.sh
+   ```
+
+3. Make the `.sh` file executable.
+
+   ```bash
+   chmod +x offline-installation-tool.sh
+   ```
+
+4. You can execute the command `./offline-installation-tool.sh -h` to see how to use the script:
+
+   ```bash
+   root@master:/home/ubuntu# ./offline-installation-tool.sh -h
+   Usage:
+   
+     ./offline-installation-tool.sh [-l IMAGES-LIST] [-d IMAGES-DIR] [-r PRIVATE-REGISTRY] [-v KUBERNETES-VERSION ]
+   
+   Description:
+     -b                     : save kubernetes' binaries.
+     -d IMAGES-DIR          : the dir of files (tar.gz) which generated by `docker save`. default: /home/ubuntu/kubesphere-images
+     -l IMAGES-LIST         : text file with list of images.
+     -r PRIVATE-REGISTRY    : target private registry:port.
+     -s                     : save model will be applied. Pull the images in the IMAGES-LIST and save images as a tar.gz file.
+     -v KUBERNETES-VERSION  : download kubernetes' binaries. default: v1.17.9
+     -h                     : usage message
+   ```
+
+5. Download the Kubernetes binary file.
+
+   ```bash
+   ./offline-installation-tool.sh -b -v v1.17.9 
+   ```
+
+   If you cannot access the object storage service of Google, run the following command instead to add the environment variable to change the source.
+
+   ```bash
+   export KKZONE=cn;./offline-installation-tool.sh -b -v v1.17.9 
+   ```
+
+   {{< notice note >}}
+
```

Comment:
For cross-version upgrades, an intermediate version also needs to be downloaded in advance.

Reply:
In Prerequisites, I have additional tips about the upgrade to v2.1.1.